rpc/src/client.rs

```diff
@@ -159,41 +173,137 @@ impl Client {
         self.perform(evidence::Request::new(e)).await
     }
 
+    /// Creates a subscription management interface for this RPC client. This
+    /// interface facilitates subscribing and unsubscribing from receiving
+    /// events produced by specific RPC queries.
+    pub async fn new_subscription_manager(
```

Comment:
It's surpising to see a constructor for a "foreign" - as in from another mod - on this entity. It might be better to have a `TryFrom` implementation on `SubscriptionManager` for `Client`.

Reply:
The trouble with the `TryFrom` approach is that it's not `async`.

We need a way to have the `SubscriptionClient` (the new interface name for the `SubscriptionManager` now) be produced from the `Client` (since in certain transports the `SubscriptionClient` will be tightly coupled to the `Client`), and doing so involves creating a WebSocket connection if you use `HttpTransport` at present, which is `async`.

Is there a good `async` alternative to `TryFrom`? I'm still working on my `async`/Rust right now 😁